scality_glance_store/store.py

```diff
@@ -117,6 +118,49 @@ def __init__(self, conf):
     def get_schemes():
         return (SCALITY_SCHEME,)
 
+    def _get_range_headers(self, image_id, offset=0, chunk_size=None):
+        """
+        Calculate range headers for partial object retrieval.
+        """
+        # Obtain object size by a HEAD request. This is to ensure compatibiliy
+        # with versions of sproxyd which has limited support for implicit
+        # ranges in the HTTP Range header.
+        size = 0
+        try:
+            headers = self._sproxyd_client.head(image_id)
```

Comment:
You should do the HEAD request if and only if `chunk_size` is None, to avoid an unnecessary http call in some case.